Makefile

```diff
@@ -1,7 +1,6 @@
-RABBITMQ_SRC_VERSION=rabbitmq_v3_2_1
+RABBITMQ_SRC_VERSION=v3.8.x
```

Comment:
To make builds more repeatable, I think this should target a specific version rather than a branch.
What might happen? Say a change to the spec is introduced between RabbitMQ 3.8.x versions; one automated build of amqplib would have a codec from before the change, and another -- say from a merge commit -- might have a codec with the change. The result would not be a _catastrophe_, but at least an avoidable cause of confusion.

Reply:
Totally agree. Though, I would prefer to change the build phase instead.

1. The `amqp-rabbitmq.json` specification and `lib/defs.js` should be git added to the repository.
   * Only maintainers usually want to update these files, they do so by calling `make clean; make lib/defs.js`.
 
   * The spec changes then can be tracked and reviewed in pull requests (but nobody will dare)
   * Npm will be able to install the package from a github branch (see #571), like
       ```
       "amqplib": "github:squaremo/amqp.node#pull/584/head"
       ```
      Now, after downloading the source files, the `prepare` npm-script is called and may fail, as not all OSs have `make` installed.
2. Prepublish npm-script or CI cron should check a mismatch of the local and remote `rabbitmq-server/master/.../amqp-rabbitmq.json`.

Reply:
Yeah I'm OK with defs.js being committed, so long as there's an automated check that it's correctly generated.

Reply:
NB I would still like to have an exact version of the spec, so that its provenance is recorded. But that still fits with your plan.

Reply:
Done. An exact commit version or a branch name or a tag can be written there.